homeassistant/components/insteon/__init__.py

```diff
@@ -47,68 +98,51 @@ async def async_setup(hass, config):
     username = conf.get(CONF_HUB_USERNAME)
     password = conf.get(CONF_HUB_PASSWORD)
     hub_version = conf.get(CONF_HUB_VERSION)
-    overrides = conf.get(CONF_OVERRIDE, [])
-    x10_devices = conf.get(CONF_X10, [])
-    x10_all_units_off_housecode = conf.get(CONF_X10_ALL_UNITS_OFF)
-    x10_all_lights_on_housecode = conf.get(CONF_X10_ALL_LIGHTS_ON)
-    x10_all_lights_off_housecode = conf.get(CONF_X10_ALL_LIGHTS_OFF)
 
     if host:
-        _LOGGER.info("Connecting to Insteon Hub on %s", host)
-        conn = await insteonplm.Connection.create(
+        _LOGGER.info("Connecting to Insteon Hub on %s:%d", host, ip_port)
+    else:
+        _LOGGER.info("Connecting to Insteon PLM on %s", port)
+
+    try:
+        if not await async_connect(
+            device=port,
             host=host,
             port=ip_port,
             username=username,
             password=password,
             hub_version=hub_version,
-            loop=hass.loop,
-            workdir=hass.config.config_dir,
-        )
-    else:
-        _LOGGER.info("Looking for Insteon PLM on %s", port)
-        conn = await insteonplm.Connection.create(
-            device=port, loop=hass.loop, workdir=hass.config.config_dir
-        )
-
-    insteon_modem = conn.protocol
+        ):
+            return True
```

Comment:
Will we retry later? Otherwise we should `return False` on failure.

Reply:
I realized this code will never be reached but the code after the exception is where I need to return False.